spec/reactors/consensus.md

```diff
@@ -0,0 +1,261 @@
+# Consensus Reactor
+
+Consensus reactor handles message propagation for 4 different channels, namely, `StateChannel`, `DataChannel`, `VoteChannel`, and `VoteSetBitsChannel`.
+The focus of this document is on the `DataChannel` and also covers the relevant parts of the `StateChannel`.
+
+## Message Types
+
+We will refer to the following message types in the following sections.
+
+### Part
+
+The `Part` serves as a representation for a block part.
+Its `bytes` field is constrained to a maximum size of [64kB](https://github.com/celestiaorg/celestia-core/blob/5a7dff4f3a5f99a4a22bb8a4528363f733177a2e/types/params.go#L19).
+`Proof` is the Merkle inclusion proof of the block part in the block (it is the proof of its inclusion in the Merkle root `Hash` found in the `PartSetHeader` of that particular block)
+```go
+type Part struct {
+Index uint32       `protobuf:"varint,1,opt,name=index,proto3" json:"index,omitempty"`
+Bytes []byte       `protobuf:"bytes,2,opt,name=bytes,proto3" json:"bytes,omitempty"`
+Proof crypto.Proof `protobuf:"bytes,3,opt,name=proof,proto3" json:"proof"`
+}
```

Comment:
[nit] the indentation of this code block appears off

Reply:
Fixed https://github.com/celestiaorg/celestia-core/pull/1129/commits/a75469838545e496c3b6dd121ce058e8946b1e7d